operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -392,20 +392,250 @@ Once this step has completed:
 
 This section describes how to update software on management nodes. It describes how to test a new image and CFS configuration on a single "canary node" first before rolling it out to the other management nodes. Modify the procedure
 as necessary to accommodate site preferences for rebuilding management nodes. The images and CFS configurations used are created by the `prepare-images` and `update-cfs-config` stages respectively; see the `prepare-images`
-[Artifacts created](../stages/prepare_images.md#artifacts-created) section for details on how to query the images and CFS configurations and [update-cfs-config](../stages/update_cfs_config.md) section for details about how the CFS configuration is updated.
+[Artifacts created](../stages/prepare_images.md#artifacts-created) section for details on how to query the images and [update-cfs-config](../stages/update_cfs_config.md) section for details about how the CFS configuration is updated.
 
 **`NOTE`** Additional arguments are available to control the behavior of the `management-nodes-rollout` stage, for example `--limit-management-rollout` and `-cmrp`. See the
 [`management-nodes-rollout` stage documentation](../stages/management_nodes_rollout.md) for details and adjust the examples below if necessary.
 
-**`NOTE`** The `management-nodes-rollout` stage creates additional separate Argo workflows when rebuilding NCN worker nodes. The Argo workflow names will include the string `ncn-lifecycle-rebuild`. If monitoring progress with the Argo UI,
-remember to include these workflows.
+**Important** There is a different procedure for  `management-nodes-rollout` depending on whether or not CSM is being upgraded.
+The two procedures differ in the handling of NCN storage nodes and NCN master nodes. If CSM is not being upgraded, then NCN storage nodes and NCN master nodes will not be upgraded and will be updated by the CFS configuration
+created in [update-cfs-config](../stages/update_cfs_config.md).
+If CSM is being upgraded, the NCN storage nodes and NCN master nodes will be upgraded.
+Both procedures use the same steps for rebuilding/upgrading NCN worker nodes.
+Select **one** of the following procedures based on whether or not CSM is being upgraded.
+
+- [Management-nodes-rollout with CSM upgrade](#651-management-nodes-rollout-with-csm-upgrade)
+- [Management-nodes-rollout with no CSM upgrade](#652-management-nodes-rollout-with-no-csm-upgrade)
+
+#### 6.5.1 Management-nodes-rollout with CSM upgrade
+
+All management nodes will need to be upgraded to a new image because CSM is being upgraded. NCN master nodes, excluding `ncn-m001`, and NCN worker nodes can be upgraded with IUF.
+NCN storage nodes and `ncn-m001` will be upgraded with manual commands.
+This section describes how to test a new image and CFS configuration on a single "canary node" for NCN master nodes and NCN worker nodes first before rolling it out to the other NCN master nodes and NCN worker nodes.
```

Comment:
This should no longer refer to worker nodes as that is in a separate section now.

Reply:
I am choosing to leave this as is. I think this is the overview of the section for upgrading all management nodes, including worker nodes. In my opinion it is not harmful to leave. It can be changed later if needed.